### PR TITLE
Add compatibilty tests against v2.5.0

### PR DESCRIPTION
**Pull request by xfxyjwf**
@pherl for review. Most of the actual code/protos are just coped from v2.5.0 source. Only these files are of interest to review:
java/compatibility_tests/README.md
java/compatibility_tests/v2.5.0/test.sh
java/compatibility_tests/v2.5.0/pom.xml
java/compatibility_tests/v2.5.0/protos/pom.xml
java/compatibility_tests/v2.5.0/more_protos/pom.xml
java/compatibility_tests/v2.5.0/tests/pom.xml

### CODE REVIEW

**Comment by xfxyjwf** [comment]
ping @pherl

File: java/compatibility_tests/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+It's also divided into 3 submodule with tests depending on more_protos and
+more_protos depending on protos. This way we can test scenarios where only part
+of the dependency is upgraded to the new version.
+
+## How to Run The Tests
+
+Before running the tests, make sure you have already built the protoc binary
+following [the C++ installation instructions](../../src/README.md). The test
+scripts will use the built binary located at ${protobuf}/src/protoc.
```
Comment (pherl) [inline java/compatibility_tests/README.md]: mention which platforms the test supports? From the script, it seems to be only tested under linux.

**Comment by pherl** [comment]
LG

File: java/compatibility_tests/README.md (hunk 1 above)
Comment (xfxyjwf) [inline java/compatibility_tests/README.md]: Done.

**Pull request merged**